homeassistant/components/unifi/switch.py

```diff
@@ -37,51 +46,224 @@
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.restore_state import RestoreEntity
 
-from .const import (
-    ATTR_MANUFACTURER,
-    BLOCK_SWITCH,
-    DOMAIN as UNIFI_DOMAIN,
-    DPI_SWITCH,
-    OUTLET_SWITCH,
-    POE_SWITCH,
-)
+from .const import ATTR_MANUFACTURER, DOMAIN as UNIFI_DOMAIN, POE_SWITCH
 from .controller import UniFiController
 from .unifi_client import UniFiClient
 
 CLIENT_BLOCKED = (EventKey.WIRED_CLIENT_BLOCKED, EventKey.WIRELESS_CLIENT_BLOCKED)
 CLIENT_UNBLOCKED = (EventKey.WIRED_CLIENT_UNBLOCKED, EventKey.WIRELESS_CLIENT_UNBLOCKED)
 
-T = TypeVar("T")
+Data = TypeVar("Data")
+Handler = TypeVar("Handler")
+
+Subscription = Callable[[CallbackType, ItemEvent], UnsubscribeType]
+
+
+def async_dpi_group_is_on_fn(
```

Comment:
Why is this `async_..`? Doesn't seem async. (Same repeats for everything else that's `async_...` but isn't.)

Reply:
I don't know if that is requested now but earlier it was expected to mark async friendly methods with async_

Reply:
Hmm. I would imagine it is (or has been) an useful hint for "you'll want to `await` this", but for these functions that's obviously not the case...

Reply:
Since Home Assistant has both sync and async code in the same code base, we prefix functions to hint that they are safe to run in async and won't do I/O with `async_` (and usually decorate them with `@callback` if they will ever be run by one of the `HassJob` runners)

Reply:
Thanks. Took a moment for me to parse "that they are safe to run in async and won't do I/O", but I take it you mean "won't do synchronous I/O"  :)